pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -284,31 +287,43 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
 
     // Calculate routing table for the query
     long routingStartTimeNs = System.nanoTime();
-    Map<ServerInstance, List<String>> offlineRoutingTable = null;
-    Map<ServerInstance, List<String>> realtimeRoutingTable = null;
+    Map<ServerInstance, List<String>> offlineServerInstanceToSegmentsMap = null;
+    Map<ServerInstance, List<String>> realtimeServerInstanceToSegmentsMap = null;
+    // Set number of segments with no replicas matched with this query
+    int unavailableSegmentCount = 0;
     if (offlineBrokerRequest != null) {
       // NOTE: Routing table might be null if table is just removed
-      offlineRoutingTable = _routingManager.getRoutingTable(offlineBrokerRequest);
-      if (offlineRoutingTable == null || offlineRoutingTable.isEmpty()) {
+      RoutingTable offlineRoutingTable = _routingManager.getRoutingTable(offlineBrokerRequest);
+      if (MapUtils.isEmpty(offlineRoutingTable.getServerInstanceToSegmentsMap())) {
         LOGGER.debug("No OFFLINE server found for request {}: {}", requestId, query);
         offlineBrokerRequest = null;
-        offlineRoutingTable = null;
+        offlineServerInstanceToSegmentsMap = null;
+      } else {
+        offlineServerInstanceToSegmentsMap = offlineRoutingTable.getServerInstanceToSegmentsMap();
+        unavailableSegmentCount += offlineRoutingTable.getSegmentsWithNoReplicas().size();
       }
     }
     if (realtimeBrokerRequest != null) {
       // NOTE: Routing table might be null if table is just removed
-      realtimeRoutingTable = _routingManager.getRoutingTable(realtimeBrokerRequest);
-      if (realtimeRoutingTable == null || realtimeRoutingTable.isEmpty()) {
+      RoutingTable realtimeRoutingTable = _routingManager.getRoutingTable(realtimeBrokerRequest);
+      if (MapUtils.isEmpty(realtimeRoutingTable.getServerInstanceToSegmentsMap())) {
         LOGGER.debug("No REALTIME server found for request {}: {}", requestId, query);
         realtimeBrokerRequest = null;
-        realtimeRoutingTable = null;
+        realtimeServerInstanceToSegmentsMap = null;
+      } else {
+        realtimeServerInstanceToSegmentsMap = realtimeRoutingTable.getServerInstanceToSegmentsMap();
+        unavailableSegmentCount += realtimeRoutingTable.getSegmentsWithNoReplicas().size();
       }
     }
     if (offlineBrokerRequest == null && realtimeBrokerRequest == null) {
       LOGGER.info("No server found for request {}: {}", requestId, query);
       _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.NO_SERVER_FOUND_EXCEPTIONS, 1);
       return BrokerResponseNative.EMPTY_RESULT;
     }
+
+    // Set number of segments with no replicas matched with this query
+    requestStatistics.setUnavailableSegmentCount(unavailableSegmentCount);
```

Comment:
Also log this in the (one) log statement that we do for every request:
```
segments(queried/processed/matched/consuming):{}/{}/{}/{}
```
Add "unavail" here